### PR TITLE
[MM-51582] Upgrade all dependencies

### PR DESCRIPTION
**Pull request by fmartingr**
#### Summary

Dependencies upgrade. See commits.

- Refactors:
  - `cloudflare-go`: Refactored modified signatures
  - `kube-aggregator`: Refactored removed `AddToScheme` method. [[ref](https://github.com/kubernetes/kube-aggregator/commit/fba84c7a2bfc4b6cd0e1bba8bd6f74ab10651002)]

Depends on: https://github.com/mattermost/mattermost-operator/pull/347

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-51582

#### Release Note

```release-note
feat: upgrade dependencies
```

### CODE REVIEW

File: internal/provisioner/backup_restore_operator_test.go [hunk 1]
```diff
@@ -461,7 +461,7 @@ func TestOperator_TriggerRestore(t *testing.T) {
 
 func setJobActiveWhenExists(t *testing.T, client v1.JobInterface, name string) {
 	ctx := context.Background()
-	err := wait.Poll(1*time.Second, 30*time.Second, func() (bool, error) {
+	err := wait.PollUntilContextTimeout(ctx, time.Second, 5*time.Second, true, func(ctx context.Context) (bool, error) {
```
Comment (mirshahriar) [inline internal/provisioner/backup_restore_operator_test.go:464]: Have you set this timeout to 5s intentionally? It was 30s.

File: internal/provisioner/exec.go [hunk 2]
```diff
@@ -122,7 +122,7 @@ func (provisioner Provisioner) ExecClusterInstallationJob(cluster *model.Cluster
 		return errors.Wrap(err, "failed to create CLI command job")
 	}
 
-	err = wait.Poll(time.Second, 10*time.Minute, func() (bool, error) {
+	err = wait.PollUntilContextTimeout(ctx, time.Second, 5*time.Second, true, func(_ context.Context) (bool, error) {
```
Comment (mirshahriar) [inline internal/provisioner/exec.go:125]: Same here. 10m to 5s ??

**Review by mirshahriar — APPROVED** [review]
LGTM. I have two comments though.

File: internal/provisioner/backup_restore_operator_test.go (hunk 1 above)
Comment (fmartingr) [reply internal/provisioner/backup_restore_operator_test.go]: Nice catch :)

**Review by gabrieljackson — APPROVED** [review]
LGTM!

**Comment by fmartingr** [comment]
Made some updates again :)

**Pull request merged**